pages/[jam].js

```diff
@@ -43,67 +38,35 @@ const Post = () => {
   }, [cookies]);
 
   const loadQuestion = () => {
-    jamsRef
-      .where('urlPath', '==', jamId)
-      .get()
-      .then((querySnapshot) => {
-        querySnapshot.forEach((doc) => {
-          let jamData = doc.data();
-          jamData.statements = {};
-
-          const statementsPromise = jamsRef
-            .doc(doc.id)
-            .collection('statements')
-            .get()
-            .then((query) => {
-              query.forEach((statement) => {
-                const statement_id = statement.id;
-                jamData.statements[statement_id] = statement.data();
-              });
-              return jamData;
-            });
-
-          const allVotes = [];
-          const votesPromise = participantsRef
-            .doc(participantId)
-            .collection('votes')
-            .get()
-            .then((query) => {
-              query.forEach((vote) =>
-                allVotes.push(vote.data().statementId),
-              );
-              return allVotes;
-            });
-
-          Promise.all([statementsPromise, votesPromise]).then(
-            ([jam, votes]) => {
-              const unansweredQs = pickBy(
-                jamData.statements,
-                (value, key) => !votes.includes(key),
-              );
-
-              const keys = Object.keys(unansweredQs);
-              if (!keys.length) {
-                setIsDone(true);
-                return;
-              }
-
-              const randomKey =
-                keys[(keys.length * Math.random()) << 0];
-              const randomQ = unansweredQs[randomKey];
-              randomQ.key = randomKey;
-
-              setQuestion(randomQ);
-            },
-          );
-        });
+    fetch(
+      `/api/question?jamId=${encodeURIComponent(
+        jamId,
+      )}&participantId=${encodeURIComponent(participantId)}`,
+    )
+      .then((response) => response.json())
+      .then((question) => {
+        const keys = Object.keys(question);
+        if (!keys.length) {
+          setIsDone(true);
+          return;
+        }
+
+        setQuestion(question);
       });
   };
 
   const settingIdCookies = () => {
-    participantsRef.add({}).then((docRef) => {
-      setCookies('jams-participant', docRef.id);
-    });
+    fetch('/api/participant', {
+      method: 'POST',
+    })
+      .then((response) => response.json())
+      .then((participant) => {
+        // Can we do this using a Set-Cookie header?
```

Comment:
would there be a difference?

Reply:
Would the `cookies` attr be updated here: https://github.com/redbadger/jams/blob/main/pages/%5Bjam%5D.js#L35-L43 through us using a `Set-Cookie` header rather than the `setCookies` hook as we're currently using it?

Reply:
I'm not sure this would work as there wouldn't be a page refresh on the UI side, so the cookies wouldn't update? Maybe one to consider later?